### PR TITLE
Upgrade to Aeron 1.19.1

### PR DESCRIPTION
**Pull request by chbatey**
No mention of on the wire incompatibilities in release notes for
versions 1.15->1.19

### CODE REVIEW

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by chbatey** [comment]
@patriknw an additional change was required so dismissed your review

File: akka-remote/src/main/resources/reference.conf [hunk 1]
```diff
@@ -1090,6 +1090,10 @@ akka {
           # Only used when transport is aeron-udp.
           client-liveness-timeout = 20 seconds
 
+          # Timout after after which an uncommitted publication will be unblocked
+          # Only used when transport is aeron-udp.
+          publication-unblock-timeout = 40 seconds
```
Comment (patriknw) [inline akka-remote/src/main/resources/reference.conf]: How did you find that this was needed? Default is 10 seconds? Why do we need a different default?

This is a property for the (embedded) media driver, and unless we have strong reasons for adding our own (different) configuration for those I think we should use defaults and users can override with `-D` System property.

File: akka-remote/src/main/scala/akka/remote/artery/aeron/ArteryAeronUdpTransport.scala [hunk 2]
```diff
@@ -95,6 +95,7 @@ private[remote] class ArteryAeronUdpTransport(_system: ExtendedActorSystem, _pro
         driverContext.aeronDirectoryName(randomName)
       }
       driverContext.clientLivenessTimeoutNs(settings.Advanced.Aeron.ClientLivenessTimeout.toNanos)
+      driverContext.publicationUnblockTimeoutNs(settings.Advanced.Aeron.PublicationUnblockTimeout.toNanos)
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/aeron/ArteryAeronUdpTransport.scala]: should probably be in `SharedMediaDriverSupport` also

**Comment by chbatey** [comment]
Every test failed :) reason is in the commit msg

The media driver won’t start with the default as we already override client
liveness timeout and this has to be bigger than that. I picked 40s as it
keeps the same ratio as the defaults

File: akka-remote/src/main/resources/reference.conf (hunk 1 above)
Comment (patriknw) [reply akka-remote/src/main/resources/reference.conf]: > Every test failed :) reason is in the commit msg
>
> The media driver won’t start with the default as we already override client
liveness timeout and this has to be bigger than that. I picked 40s as it
keeps the same ratio as the defaults

ok, good

**Review by patriknw — APPROVED** [review]
LGTM after adding to SharedMediaDriverSupport

File: akka-remote/src/main/scala/akka/remote/artery/aeron/ArteryAeronUdpTransport.scala (hunk 2 above)
Comment (chbatey) [reply akka-remote/src/main/scala/akka/remote/artery/aeron/ArteryAeronUdpTransport.scala]: Yep, adding now

File: akka-remote/src/main/scala/akka/remote/artery/ArterySettings.scala [hunk 3]
```diff
@@ -203,6 +203,10 @@ private[akka] final class ArterySettings private (config: Config) {
         config
           .getMillisDuration("client-liveness-timeout")
           .requiring(interval => interval > Duration.Zero, "client-liveness-timeout must be more than zero")
+      val PublicationUnblockTimeout: FiniteDuration =
+        config
+          .getMillisDuration("publication-unblock-timeout")
+          .requiring(interval => interval > Duration.Zero, "publication-unblock-timeout must be more than zero")
```
Comment (helena) [inline akka-remote/src/main/scala/akka/remote/artery/ArterySettings.scala]:
```suggestion
          .requiring(interval => interval > Duration.Zero, "publication-unblock-timeout must be greater than zero")
```

File: akka-remote/src/main/resources/reference.conf (hunk 1 above)
Comment (helena) [reply akka-remote/src/main/resources/reference.conf]: I am wondering the same, but I see it in the commit.

**Review by helena — APPROVED** [review]
Looks good

**Pull request merged**